### DISCUSSION TITLE
EMS-ESP Problems with "heating temperature" and "selected flow temperature"

### DISCUSSION THREAD

**Discussion by MarkRoks** [emsesp/EMS-ESP32#975] [General]
Hello,

I use the EMS-ESP gateway to control/read out the Nefit Trendline. I use the Nefit Easy thermostat for the regulation.

I set the maximum supply temperature by means of "heating temperature". However, if I adjust this, the "selected flow temperature" is also adjusted, to the same value, regardless of height/reality. In addition, the "selected flow temperature" is not adjustable and always follows the "heating temperature".

This applies to external adjustments (Open API), as well as via the web page of the EMS-ESP gateway and via MQTT. I am using version v3.3.1 as firmware for the EMS-ESP gateway.

When I use the Open API, I see the following in the gateway's SystemLog:

**2022-02-15 10:04:37.268    INFO 19835:  [command]    Calling boiler command 'selFlowTemp', value 40, id is 1
2022-02-15 10:04:37.268    INFO 19836:  [boiler]     Setting boiler flow temperature to 40 C
2022-02-15 10:04:37.268   DEBUG 19837:  [emsesp]     API command called successfully**

I don't get any errors, but nothing happens according to the Dashboard.

If I use MQTT, I see the following in the gateway's SystemLog:

**2022-02-15 10:12:16.619    INFO 20643:  [command]    Calling boiler command 'selFlowTemp', value 35, id is 1
2022-02-15 10:12:16.619    INFO 20644:  [boiler]     Setting boiler flow temperature to 35 C**

And also I don't get any errors, but nothing happens according to the Dashboard.

Where is this going wrong, or am I doing it wrong?

Regards,

Mark

**Comment by MichaelDvP** [comment]
What do you want to do? You have a thermostat. The thermostat calcualtes the required flowtemp to achieve the right roomtemperature. If you set a flowtemp via emsesp the active thermostat will overwrite it a few seconds later. 
But why you are changing the flowtemp if you have a thermostat?

**Comment by MarkRoks** [comment]
Hi Michael,

What I would like to do (and has always worked) is to send a self-determined "flow temperature" to the boiler. This is determined by a tool (in Node-Red), based on the measured outside and inside temperature. And this has worked in the past.

Apart from that, the "flow temperature" should not just take over the "heating temperature". Now these are always identical and nothing is adjusted from the thermostat when I deactivate my tool. In short, both values ​​remain the same, regardless of what the thermostat measures (steer like this).

If the "heating temperature" has been adjusted and it is automatically/completely taken over by the "flow temperature", the thermostat will no longer adjust this to the period after.

In addition, the target values ​​in the EMS-ESP Dashboard can also be adjusted manually. However, the "flow temperature" immediately takes over the "heating temperature". The very strange thing about this is, also the other way around, if you adjust the "flow temperature" here.....

What I would like is to determine a maximum limit value "heating temperature" (fixing) and be able to determine what the boiler should do with the "flow temperature" separately from this.

**Comment by MichaelDvP** [comment]
Sorry i don't understand. If you don't have a mixer flowtemp and heatingtemp are idencal, there is no other way to reach the flowtemp as heating. With a mixer the heatingtemp can be higher and the mixer reduces the flowtemp. 
If there is a thermostat you can not set the boiler flowtemp, you have to communicate to thermostat.

**Comment by MarkRoks** [comment]
Hello michael,

Maybe I didn't explain it enough. This is because it concerns the values ​​to be adjusted;
- Selected flow temperature
- Heating temperature

Where the "heating temperature" is normally fixed on the boiler, as being the upper limit. The desired boiler water temperature would then be controlled by means of the "Selected flow temperature". This can of course never be higher than the "heating temperature". The same is of course possible, but this would only be possible if the "heating temperature" is set low and/or there is a lot of heat demand "Selected flow temperature".

Both values ​​can therefore be adjusted manually in the EMS-ESP dashboard, although the thermostat could of course also determine the "Selected flow temperature".

However, they are now always identical, which theoretically should not be possible. Certainly not, if you adjust 1 of the two, the other will (automatically) be adjusted immediately, to the same value. Why would you be able to adjust them (independently of each other) in the dashboard?

Apart from all this, I actually want to overrule the control from the thermostat with this, towards the boiler.

**Comment by mvjt** [comment]
I´m also interested in controlling the flow temperature of my heat pump. What does "hc1 target flow temperature" mean? It is 0 degrees on my HP...

**Comment by MichaelDvP** [comment]
> I´m also interested in controlling the flow temperature of my heat pump. 

Why?
> What does "hc1 target flow temperature" mean? It is 0 degrees on my HP...

There is no "hc1 target flow temperature" on the HP. It is a thermostat value. It is the calculated flowtemp setpoint send to the boiler/HP. If you want to change it, change the setpoint of roomtemp or change the heatingcurve and the thermostat calculates a new target flow temp. But you can't set the flowtemp on boiler/HP if you have a thermostat.

**Comment by MarkRoks** [comment]
Hey,

Managing the selected flow temperature arose from the fact that it turned out to be adjustable within EMS-ESP, just like the heating temperature. In this way I made my own heating curve calculation, which works perfectly. Regardless of whether the thermostat would want to control something else.

However, within the possibilities of the EMS-ESP gateway it appears not possible to only adjust the selected flow temperature. By adjusting the heating temperature, the selected flow temperature is of course also adjusted. It is therefore possible to control the boiler in this way.

However, the option to adjust the selected flow temperature in this way has no added value. And I control the maximum heating temperature within the script with which I now determine the heating curve.

So I'm still wondering why the possibility exists to adjust the selected flow temperature, since it adds "nothing"?

**Comment by MarkRoks** [comment]
Hey,

After some research, I found out about the heating temperature. It was noticed that the 2 available thermostats (Nefit and Bosch) work differently compared to each other.

Since I first used the Nefit Easy thermostat, transferring the heating temperature to the boiler went perfectly, as reported.

I have now tried the same with the Bosch EasyControl, but this thermostat does not accept that a different value is sent to the boiler. In other words, the controlled heating temperature is overruled by the thermostat. In addition, the minimum step size is 0.5 K with the Bosch, he quickly controls the heating temperature to 5 degrees, by rounding up the target value if it gets too hot.

With the Nefit Easy, 0.1 K steps are used. Because this works much more finely, a changed heating temperature is controlled more realistically and is therefore easier to correct and is therefore also accepted.

In short, what I have made does not work with the Bosch EasyControl, but it does with the Nefit Easy.

Does anyone have similar experiences, with either the Bosch or the Nefit (or maybe both, just like me).

Regards,

**Comment by marcinwrobel** [comment]
> Hey,
> 
> ... I made my own heating curve calculation, which works perfectly. Regardless of whether the thermostat would want to control something else....
> 

Hi, show me your heating curve template please :)
